### ISSUE TITLE
Subcommand with inherited mixed-in standard help options + HelpCommand class: failure

### ISSUE THREAD

**Issue by deining** [remkop/picocli#1319]
**Minimum working example:**

```
import picocli.CommandLine;
import picocli.CommandLine.ScopeType;
import picocli.CommandLine.*;

@Command(scope = ScopeType.INHERIT, mixinStandardHelpOptions = true, subcommands = { HelpCommand.class })
public class InheritHelpApp  {

    @Command()
    void sub(@Option(names = "-foo") int foo) {
        System.out.printf("Foo: %d", foo);
    }

    public static void main (String... args) {
        new CommandLine(new InheritHelpApp()).execute("sub", "-foo", "42" );
    }
}
```

When running this code, I'm getting

```
Exception in thread "main" picocli.CommandLine$DuplicateOptionAnnotationsException: Option name '-h' is used by both field boolean picocli.CommandLine$AutoHelpMixin.helpRequested and field boolean picocli.CommandLine$HelpCommand.helpRequested
	at picocli.CommandLine$DuplicateOptionAnnotationsException.create(CommandLine.java:17971)
	at picocli.CommandLine$DuplicateOptionAnnotationsException.access$2500(CommandLine.java:17965)
	at picocli.CommandLine$Model$CommandSpec.addOption(CommandLine.java:6574)
	at picocli.CommandLine$Model$CommandSpec.addMixin(CommandLine.java:6820)
	at picocli.CommandLine$Model$CommandSpec.mixinStandardHelpOptions(CommandLine.java:7162)
	at picocli.CommandLine$Model$CommandSpec.mixinStandardHelpOptions(CommandLine.java:7178)
	at picocli.CommandLine$Model$CommandReflection.extractCommandSpec(CommandLine.java:11409)
	at picocli.CommandLine$Model$CommandSpec.forAnnotatedObject(CommandLine.java:6204)
	at picocli.CommandLine.<init>(CommandLine.java:227)
	at picocli.CommandLine.<init>(CommandLine.java:221)
	at picocli.CommandLine.<init>(CommandLine.java:196)
	at InheritHelpApp.main(InheritHelpApp.java:14)
```

It's the combination of inherited `scope` + `mixinStandardHelpOptions`  + `subcommands = { HelpCommand.class }` that's causing trouble here. If I set `scope` to `ScopeType.LOCAL`, the sample runs fine.

**Comment by remkop** [comment]
Good catch!

I thought I had fixed this with #1316, but it turns out that the `HelpCommand` has a `-h`/`--help` option that does not use `mixinStandardHelpOptions`... 

I may need to add special logic for `HelpCommand`, still thinking about how to address this.

**Comment by remkop** [comment]
I finally had time to look at this.

I resolved this by not inheriting the standard help options mixin if a command already has one of the `-h`, `--help`, `-V` or `--version` options.
